cfsdk/src/main/java/io/harness/cfsdk/SdkThread.java

```diff
@@ -65,6 +67,8 @@ class SdkThread implements Runnable {
     private final Set<EventsListener> eventsListenerSet;
     private final AuthCallback authCallback;
     private final NetworkChecker networkChecker;
+    private final AtomicBoolean running;
```

Comment:
This is a fallback, in case the thread interruption does not work as expected. It is used in the `Run()` while loop condition.